autoload/mundo.vim

```diff
@@ -385,28 +388,88 @@ function! mundo#MundoHide()"{{{
 endfunction"}}}
 
 function! mundo#MundoRenderGraph()"{{{
+    " Save window view information and buffer number
+    let currentWin  = bufwinnr('%')
+    let winView = winsaveview()
+
     call s:MundoPython('MundoRenderGraph()')
+
+    " Return to buffer and restore window view
+    execute currentWin .'wincmd w'
+    call winrestview(winView)
 endfunction"}}}
 
 " automatically reload Mundo buffer if open
 function! s:MundoRefresh()"{{{
-  " abort when there were no changes
+    " abort if Mundo is closed or cursor is in the preview window
+    let mundoWin    = bufwinnr('__Mundo__')
+    let mundoPreWin = bufwinnr('__Mundo_Preview__')
+    let currentWin  = bufwinnr('%')
+
+    if (mundoWin == -1) || (mundoPreWin == -1) || (mundoPreWin == currentWin)
+        return
+    endif
+
+    " Normal refresh
+    if !get(g:, 'mundo_auto_preview', 0) ||
+                \ get(g:, 'mundo_auto_preview_delay', 0) <= 0
+        return mundo#MundoRenderGraph()
+    endif
 
-  let mundoWin    = bufwinnr('__Mundo__')
-  let mundoPreWin = bufwinnr('__Mundo_Preview__')
-  let currentWin  = bufwinnr('%')
+    " Delayed refresh
+    if !get(g:, 'mundo_enable_inline_delay', 0) ||
+                \ !get(g:, 'mundo_inline_undo')
+        call mundo#MundoRenderGraph()
+    endif
+
+    call s:MundoRestartRefreshTimer()
+endfunction"}}}
 
-  " abort if Mundo is closed or is current window
-  if (mundoWin == -1) || (mundoPreWin == -1) || (mundoPreWin == currentWin)
-    return
-  endif
+function! s:MundoRestartRefreshTimer()"{{{
+    call s:MundoStopRefreshTimer()
+    let s:auto_preview_timer = timer_start(
+                \ get(g:, 'mundo_auto_preview_delay', 0),
+                \ function('s:MundoRefreshDelayed')
+                \ )
+endfunction"}}}
+
+function! s:MundoStopRefreshTimer()"{{{
+    if s:auto_preview_timer != -1
+        call timer_stop(s:auto_preview_timer)
+        let s:auto_preview_timer = -1
+    endif
+endfunction"}}}
 
-  let winView = winsaveview()
-  :MundoRenderGraph
+function! s:MundoRefreshDelayed(...)"{{{
+    " abort if Mundo is closed or cursor is in the preview window
+    let mundoWin    = bufwinnr('__Mundo__')
+    let mundoPreWin = bufwinnr('__Mundo_Preview__')
+    let currentWin  = bufwinnr('%')
 
-  " switch back to previous window
-  execute currentWin .'wincmd w'
-  call winrestview(winView)
+    if (mundoWin == -1) || (mundoPreWin == -1) || (mundoPreWin == currentWin)
+        return
+    endif
+
+    " Handle other windows
+    if currentWin != mundoWin
+        return mundo#MundoRenderGraph()
+    endif
+
+    " Handle graph window (__Mundo__)
+    if s:auto_preview_line == line('.')
+        return
+    endif
+
+    if mode() != 'n'
+        return s:MundoRestartRefreshTimer()
+    endif
+
+    if get(g:, 'mundo_enable_inline_delay', 0) && get(g:, 'mundo_inline_undo')
```

Comment:
As mentioned in the comments above, I suggest removing the `mundo_enable_inline_delay` part of this if-block.